packages/video_player/video_player_avfoundation/ios/Classes/FLTVideoPlayerPlugin.m

```diff
@@ -319,11 +319,17 @@ - (void)observeValueForKeyPath:(NSString *)path
     }
   } else if (context == playbackBufferEmptyContext) {
     if (_eventSink != nil) {
-      _eventSink(@{@"event" : @"bufferingStart"});
+      if ([[_player currentItem] isPlaybackLikelyToKeepUp]) {
+        _eventSink(@{@"event" : @"bufferingEnd"});
+      } else {
+        _eventSink(@{@"event" : @"bufferingStart"});
+      }
     }
   } else if (context == playbackBufferFullContext) {
     if (_eventSink != nil) {
-      _eventSink(@{@"event" : @"bufferingEnd"});
+      if ([[_player currentItem] isPlaybackLikelyToKeepUp]) {
```

Comment:
from the [doc](https://developer.apple.com/documentation/avfoundation/avplayeritem/1390348-playbacklikelytokeepup?language=objc), when buffer is full, it's still possible for `isPlaybackLikelyToKeepUp` to return NO. Do we want to emit `bufferingStart` then?

Reply:
Yes, that is correct. Since the KVO code related to 'playbackBufferEmptyContext' is already responsible for emitting the 'bufferingStart' event, there is no need to emit it here.

Reply:
I believe I may have overlooked adding the same logic to the `playbackBufferEmptyContext`. As stated in the [doc](https://developer.apple.com/documentation/avfoundation/avplayeritem/1390348-playbacklikelytokeepup?language=objc#:~:text=A%20Boolean%20value%20that%20indicates%20whether%20the%20item%20will%20likely%20play%20through%20without%20stalling.) the isPlaybackLikelyToKeepUp property is a Boolean value that indicates whether the item will likely play through without stalling. Already install the new [commit](https://github.com/flutter/packages/pull/3826/commits/c50d9e20149f93e8330a26dd35ae71a7aa514922).